Providers/Modules/nxOMSAuditdPlugin/Scripts/OMSAuditdPlugin.sh

```diff
@@ -0,0 +1,325 @@
+        BEGIN {
+            emit = 1;
+            start_found = 0;
+            end_found = 0;
+            count = 0;
+            idx = 0;
+        }
+        {
+            if (match($0, "^\\s*-e")) { if (idx == 0) {idx = count } next; }
+            if ($0 == "#### START OF OMS AUDIT RULES ####") { emit = 0; start_found = 1; if (idx == 0) { idx = count;} next; }
+            if ($0 == "#### END OF OMS AUDIT RULES ####") { emit = 1; end_found = 1; next; }
+            if (emit == 1) { lines[count++] = $0; next; }
+        }
+        END {
+            if (start_found != end_found) { exit 1; }
+            if (idx == 0) { idx = count; }
+            for (i = 0; i < idx; i++) { printf "%s\n", lines[i]; }
+            print "#### START OF OMS AUDIT RULES ####"
+            while(getline line<DSC_AUDIT_RULES_FILE){print line}
+            print "#### END OF OMS AUDIT RULES ####"
+            for (i = idx; i < count; i++) { printf "%s\n", lines[i]; }
+        }' > $TmpOutFile
+        if [ $? -ne 0 ]; then
+            echo "Failed to add/replace OMS audit rules to $AUDIT_RULES_FILE" >&2
+            rm -f $TmpOutFile
+            return 1
+        fi
+        cp $TmpOutFile $AUDIT_RULES_FILE
+        if [ $? -ne 0 ]; then
+            echo "Failed to edit $AUDIT_RULES_FILE" >&2
+            rm -f $TmpOutFile
+            return 1
+        fi
+        rm -f $TmpOutFile
+    fi
+}
+
+case $1 in
+    get)
+        AUDIT_VERSION=$(/sbin/auditctl -v | sed 's/^[^0-9]*\([0-9]\.[0-9]\).*$/\1/')
+        if [ $? -ne 0 ]; then
+            echo "Failed to determine auditctl version"
+            exit 2
+        fi
+        # $2 tmp dir
+        get_plugin_state > $2/auditd_plugin.state
+        if [ $? -ne 0 ]; then
+            rm $2/auditd_plugin.state
+            exit 3
+        else
+            chown omsagent.omiusers $2/auditd_plugin.state 2>&1 >/dev/null
+        fi
+
+        get_actual_rules > $2/auditd_plugin.rules
+        if [ $? -ne 0 ]; then
+            rm $2/auditd_plugin.rules
+            exit 4
+        else
+            chown omsagent.omiusers $2/auditd_plugin.rules 2>&1 >/dev/null
+        fi
+
+        /sbin/auditctl -l > $2/auditd_loaded.rules 2>/dev/null
+        if [ $? -ne 0 ]; then
+            rm $2/auditd_loaded.rules
+            exit 5
+        else
+            chown omsagent.omiusers $2/auditd_loaded.rules 2>&1 >/dev/null
+        fi
+
+        echo $AUDIT_VERSION
+        ;;
+    set)
+        # $2 restart or ""
+        # $3 plugin state (or "" if it doesn't need to change)
+        # $4 rules file path ("" if no change, "remove" to remove rules)
+        # $5 dest auoms outconf file name
+        # $6 source auoms outconf file ("" if no change, or "remove" to remove file)
+        # $7 rules to pass to auditctl -R
+        if [ "$2" == "restart" ]; then
+            /opt/microsoft/omsagent/bin/service_control restart
+            if [ $? -ne 0 ]; then
+                exit 2
```

Comment:
Each of these exit codes seem to mean something; can you move them to the top of the file as constants for easier grouping/identification?